dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -0,0 +1,332 @@
+                if to_external_source:
+                    kwargs_preprocessed[key] = _fn.external_source(
+                        name=f'{self._es_kwarg_name}{key}', device=device)
+                else:
+                    kwargs_preprocessed[key] = value
+
+            res = op_wrapper(*inputs_preprocessed, **kwargs_preprocessed)
+
+            return tuple(res) if isinstance(res, list) else res
+
+        self._external_source_debug = False
+        p = pipe()
+        p.build()
+        self._external_source_debug = True
+        return p
+
+    def _external_source(self, op_wrapper, name, *inputs, **kwargs):
+        # TODO(ksztenderski): Possibly remove this wrapper to avoid running data through the backend and back.
+        @_pipeline.pipeline_def(**self._subpipeline_kwargs)
+        def external_source_pipeline():
+            res = op_wrapper(*inputs, name=name, **kwargs)
+            return tuple(res) if isinstance(res, list) else res
+
+        self._cur_subpipeline_id += 1
+        key = inspect.getframeinfo(
+            inspect.currentframe().f_back.f_back)[:3] + (self._cur_subpipeline_id,)
+        if not self._subpipelines_built:
+            self._external_source_debug = False
+            p = external_source_pipeline()
+            p.build()
+
+            # feed_input all data collected after build and before run
+            for (data, fi_kwargs) in self._feed_input_data.pop(name, []):
+                p.feed_input(name, data, **fi_kwargs)
+
+            self._subpipelines[key] = p
+            self._external_source_pipelines[name] = p
+            self._external_source_debug = True
+
+        if key in self._subpipelines:
+            return self._run_subpipeline(self._subpipelines[key], (), {})
+        else:
+            raise RuntimeError(f'Unexpected operator {op_wrapper.__name__}. Debug mode does not support'
+                               ' changing the order of operators executed within the pipeline.')
+
+    def _run_subpipeline(self, pipe, inputs, kwargs):
+        """Run pipeline wrapper of a single operator."""
+        self._debug_on = False
+
+        for i, input in enumerate(inputs):
+            to_external_source, _, data = _PipelineDebug._classify_data(input)
+            if to_external_source:
+                pipe.feed_input(f'{self._es_input_name}{i}', data)
+        for key, value in kwargs.items():
+            to_external_source, _, data = _PipelineDebug._classify_data(value)
+            if to_external_source:
+                pipe.feed_input(f'{self._es_kwarg_name}{key}', data)
+
+        res = pipe.run()
+        res = tuple([DataNodeDebug(tensor, **data_node.__dict__)
+                    for tensor, data_node in zip(res, pipe._graph_out)])
+
+        self._debug_on = True
+
+        if len(res) == 1:
+            return res[0]
+
+        return res
+
+    def _wrap_op_call(self, op_wrapper, *inputs, **kwargs):
```

Comment:
As suggested in fn.py, maybe just 
```suggestion
    def _wrap_op_call(self, op_wrapper, inputs, kwargs):
```
and pass the list of inputs and dict with args without unpacking. Same for _create_subpipeline. Just in case of some argument name collisions?

Reply:
done